src/systems/light.js

```diff
@@ -73,5 +98,17 @@ module.exports.System = registerSystem('light', {
     sceneEl.appendChild(directionalLight);
 
     this.defaultLights = true;
+  },
+
+  /**
+   * Enables/disables the renderer shadow map.
+   * @param {boolean} enabled
+   */
+  setShadowMapEnabled: function (enabled) {
+    var renderer = this.sceneEl.renderer;
+    this.shadowMapEnabled = enabled;
+    if (renderer) {
+      renderer.shadowMap.enabled = enabled;
+    }
```

Comment:
This is a general material issue.

This is necessary if we dynamically turn on/off shadow after scene is rendered once.
So it's kinda rare situation and maybe it isn't written in any documents.

Shadow related GLSL code is kinda hard coded depending on shadow parameters.
So we need to set `material.needsUpdate` `true` to update GLSL code
if we dynamically switch `shadowMap.enable` on/off.

Probably they'll help you understand.
https://github.com/mrdoob/three.js/blob/93bfc79da1758d4891f27f809598f836728b009b/src/renderers/webgl/WebGLPrograms.js#L190
https://github.com/mrdoob/three.js/blob/ddd8c2b1eb8a92a344b261c2125d7a2997248309/src/renderers/webgl/WebGLProgram.js#L358